service/history/api/updateactivityoptions/api_test.go

```diff
@@ -0,0 +1,395 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package updateactivityoptions
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	activitypb "go.temporal.io/api/activity/v1"
+	commonpb "go.temporal.io/api/common/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	workflowservicepb "go.temporal.io/api/workflowservice/v1"
+	historyspb "go.temporal.io/server/api/history/v1"
+	"go.temporal.io/server/api/historyservice/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common/cluster"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/namespace"
+	"go.temporal.io/server/common/persistence/versionhistory"
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/events"
+	"go.temporal.io/server/service/history/hsm"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tests"
+	"go.temporal.io/server/service/history/workflow"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+func TestApplyActivityOptionsAcceptance(t *testing.T) {
+
+	fullActivityInfo := &persistencespb.ActivityInfo{
+		TaskQueue:               "task_queue_name",
+		ScheduleToCloseTimeout:  durationpb.New(time.Second),
+		ScheduleToStartTimeout:  durationpb.New(time.Second),
+		StartToCloseTimeout:     durationpb.New(time.Second),
+		HeartbeatTimeout:        durationpb.New(time.Second),
+		RetryBackoffCoefficient: 1.0,
+		RetryInitialInterval:    durationpb.New(time.Second),
+		RetryMaximumInterval:    durationpb.New(time.Second),
+		RetryMaximumAttempts:    5,
+	}
+
+	allOptions := &activitypb.ActivityOptions{
+		TaskQueue:              &taskqueuepb.TaskQueue{Name: "task_queue_name"},
+		ScheduleToCloseTimeout: durationpb.New(time.Second),
+		StartToCloseTimeout:    durationpb.New(time.Second),
+		ScheduleToStartTimeout: durationpb.New(time.Second),
+		HeartbeatTimeout:       durationpb.New(time.Second),
+		RetryPolicy: &commonpb.RetryPolicy{
+			MaximumInterval:    durationpb.New(time.Second),
+			MaximumAttempts:    5,
+			BackoffCoefficient: 1.0,
+			InitialInterval:    durationpb.New(time.Second),
+		},
+	}
+
+	testCases := []struct {
+		name     string
+		options  *activitypb.ActivityOptions
+		ai       *persistencespb.ActivityInfo
+		expected *persistencespb.ActivityInfo
+		mask     *fieldmaskpb.FieldMask
+	}{
+		{
+			name:     "full mix - CamelCase",
+			options:  allOptions,
+			ai:       &persistencespb.ActivityInfo{},
+			expected: fullActivityInfo,
+			mask: &fieldmaskpb.FieldMask{
+				Paths: []string{
+					"TaskQueue.Name",
```

Comment:
Do you think we want to support all cases combination? What is best practices here? I would start with names from proto files, but I don't have strong opinion here.

Reply:
in short - yes. (we also support "taskQueue.name" btw). This is because both are legit ways, and we may be in a situation when we don't control how those field masks are created